tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
+        #   qml.RY(f, wires=2)
+        #
+        # Note: since this layer only consists of a single parameter,
+        # only need to compute a single diagonal element.
+
+        @qml.qnode(dev)
+        def layer2_diag(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            qml.RY(f, wires=2)
+            return qml.var(qml.PauliX(1))
+
+        G2 = layer2_diag(x, y, z, h, g, f)/4
+        assert np.allclose(G[3:4, 3:4], G2, atol=tol, rtol=0)
+
+        # =============================================
+        # Test block diag metric tensor of third layer is correct.
+        # We do this by computing the required expectation values
+        # numerically using multiple circuits.
+        # The second layer includes the non_parametrized_layer,
+        # RX, RY, RZ gates (x, y, z params), and a 2nd non_parametrized_layer.
+        #
+        # Observables are the generators of:
+        #   qml.RY(f, wires=1)
+        #   qml.RZ(g, wires=2)
+        G3 = np.zeros([2, 2])
+
+        @qml.qnode(dev)
+        def layer3_diag(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            return qml.var(qml.PauliZ(2)), qml.var(qml.PauliY(1))
+
+        @qml.qnode(dev)
+        def layer3_off_diag_first_order(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            return qml.expval(qml.PauliZ(2)), qml.expval(qml.PauliY(1))
+
+        @qml.qnode(dev)
+        def layer3_off_diag_second_order(x, y, z, h, g, f):
+            non_parametrized_layer(a, b, c)
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.RZ(z, wires=2)
+            non_parametrized_layer(a, b, c)
+            return qml.expval(qml.Hermitian(np.kron(Z, Y), wires=[2, 1]))
+
+        # calculate the diagonal terms
+        varK0, varK1 = layer3_diag(x, y, z, h, g, f)
+        G3[0, 0] = varK0/4
+        G3[1, 1] = varK1/4
+
+        # calculate the off-diagonal terms
+        exK0, exK1 = layer3_off_diag_first_order(x, y, z, h, g, f)
+        exK01 = layer3_off_diag_second_order(x, y, z, h, g, f)
+
+        G3[0, 1] = (exK01 - exK0*exK1)/4
+        G3[1, 0] = (exK01 - exK0*exK1)/4
+
+        assert np.allclose(G[4:6, 4:6], G3, atol=tol, rtol=0)
+
+        # ============================================
+        # Finally, double check that the entire metric
+        # tensor is as computed.
+
+        G_expected = block_diag(G1, G2, G3)
+        assert np.allclose(G, G_expected, atol=tol, rtol=0)
+
+    def test_evaluate_diag_approx_metric_tensor(self, tol):
```

Comment:
It seems like some of the circuits here are shared with the function above? Does it make sense to separate it out of the tests? (also might enable easier test-making for related functions in future)

Reply:
Done, separated it out as a class fixture :+1: